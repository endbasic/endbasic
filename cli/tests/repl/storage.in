' EndBASIC
' Copyright 2021 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Interactive session that manipulates storage drives.

' Show default state.
MOUNT
DIR

' Load demo content before unmounting the drive, as we will use it later to
' write some new files.
LOAD "demos:/hello.bas"

' Replace some default drives with two more.
MOUNT "A", "memory://"
MOUNT "B", "memory://"
CD "A:/"
UNMOUNT "DEMOS"
UNMOUNT "MEMORY"
MOUNT

' Save files in the different drives, using absolute and relative paths, and
' check the contents.
SAVE "a:/hello1.bas"
SAVE "b:hello2.bas"
NEW
SAVE "empty1.bas"
SAVE "a:empty2.bas"
SAVE "b:/empty3.bas"
DIR
DIR "a:/"
DIR "b:"
