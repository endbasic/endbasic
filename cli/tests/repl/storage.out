
    Welcome to EndBASIC X.Y.Z.

    Type HELP for interactive usage information.
    For a guided tour, type: LOAD "DEMOS:/TOUR.BAS": RUN


    Name      Target
    DEMOS     demos://
    LOCAL     file:///PATH/TO/TMPDIR
    MEMORY    memory://

    3 drive(s)


    Directory of LOCAL:/

    Modified              Size    Name
    0 file(s), 0 bytes


    Name     Target
    A        memory://
    B        memory://
    LOCAL    file:///PATH/TO/TMPDIR

    3 drive(s)


    Directory of A:/

    Modified              Size    Name
    YYYY-MM-DD HH:MM         0    empty1.bas
    YYYY-MM-DD HH:MM         0    empty2.bas
    YYYY-MM-DD HH:MM       651    hello1.bas

    3 file(s), 651 bytes


    Directory of A:/

    Modified              Size    Name
    YYYY-MM-DD HH:MM         0    empty1.bas
    YYYY-MM-DD HH:MM         0    empty2.bas
    YYYY-MM-DD HH:MM       651    hello1.bas

    3 file(s), 651 bytes


    Directory of B:/

    Modified              Size    Name
    YYYY-MM-DD HH:MM         0    empty3.bas
    YYYY-MM-DD HH:MM       651    hello2.bas

    2 file(s), 651 bytes

End of input by CTRL-D
